pypeit/reduce.py

```diff
@@ -307,11 +309,15 @@ def global_skysub(self, sciimg, sciivar, tilts, std=False, skymask=None, update_
                 self.maskslits[slit] = True
 
         if update_crmask:
-            self.crmask = processimages.ProcessImages.build_crmask(self.sciimg - self.global_sky, self.proc_par,
-                                                                   self.det, self.spectrograph, ivar = self.sciivar,
-                                                                   binning=self.binning)
+            processImage = processimage.ProcessImage(None, self.spectrograph, self.det, self.proc_par)
```

Comment:
I do not like that we have to instantiate this class to simply run one method, build_crmask, which amounts to one line of code. The only dependency of build_crmask on processimages is the parset for image processing. This was originally set to be a class method for that reason, to avoid having to instantiate the class. Right now what was one line of code has become 5 lines of code. Perhaps this is a matter of taste, since you like many incantations and I like long argument lists. Maybe the solution is for the cr masking to have its own parameter set, and for that to be simply run avoiding instantiating processimages.

Reply:
why do you say 5 lines of code?

Reply:
Anyways, I can get it down to 2 lines if that is very much preferred.